File: src/main/java/es/uji/sape/model/InternshipOffer.java
package es.uji.sape.model;

import lombok.Data;
import lombok.NoArgsConstructor;
import org.jetbrains.annotations.NotNull;

@Data
@NoArgsConstructor
public final class InternshipOffer {
    private int id;
    private int degree;
    private @NotNull String description = "";
    private @NotNull Month startDate = Month.INDIFFERENT;
    private @NotNull String contactUsername = "";
}

File: src/main/kotlin/es/uji/sape/model/InternshipOffer.kt
package es.uji.sape.model

import java.time.LocalDate

data class InternshipOffer(
    val id: Int = 0,
    val degree: Int = 0,
    val tasks: String = "",
    val startDate: LocalDate = LocalDate.now(),
    val contactUsername: String = ""
)
